blaze/server/spider.py

```diff
@@ -72,28 +73,47 @@ def data_spider(path,
     """
     # NOTE: this is named `data_spider` rather than just `spider` to
     # disambiguate this function from the `blaze.server.spider` module.
-    return {
-        os.path.basename(path): _spider(path, ignore=ignore,
-                                        followlinks=followlinks,
-                                        hidden=hidden,
-                                        extra_kwargs=extra_kwargs)
-    }
+    return {os.path.basename(path): _spider(path,
+                                            ignore=ignore,
+                                            followlinks=followlinks,
+                                            hidden=hidden,
+                                            extra_kwargs=extra_kwargs)}
 
 
-def from_yaml(path, ignore=(ValueError, NotImplementedError), followlinks=True,
-              hidden=False):
+@contextmanager
+def manage_chdir(root):
+    """Context manager that changes to ``root`` directory on ``__entry__`` and
+    changes back to ``os.getcwd()`` on __exit__.
+    """
+    root = os.path.abspath(root)
+    cwd = os.path.abspath(os.getcwd())
+    if root != cwd:
+        os.chdir(root)
+    yield
```

Comment:
this should be in a try/finally so that if an exception is raised in the `manage_chdir` context we still correctly exit.